Detail getting started from scratch with step by step instructions

Krish1979 opened the pull request:
To provide step by step instructions for starting new Quorum nodes with both raft and IBFT consensus and with or without privacy manager..

Krish1979 committed:
Detail getting started from scratch with step by step instructions

Krish1979 committed:
Merge branch 'master' into fromscratch-docs-update

libby commented on docs/Getting Started/Getting-Started-From-Scratch.md:
```diff
@@ -1,108 +1,1150 @@
+     Your new account is locked with a password. Please give a password. Do not forget this password.
+     Passphrase: 
+     Repeat passphrase: 
+     Address: {679fed8f4f3ea421689136b25073c6da7973418f}
```
maybe add:
```
# the keystore file includes the address in the last part of its filename.
$ ls new-node-1/keystore
UTC--2019-06-17T09-29-06.665107000Z--679fed8f4f3ea421689136b25073c6da7973418f
```

libby replied:
maybe add:
```
# the keystore file includes the address in the last part of its filename.
$ ls new-node-1/keystore
UTC--2019-06-17T09-29-06.665107000Z--679fed8f4f3ea421689136b25073c6da7973418f
```

libby commented on docs/Getting Started/Getting-Started-From-Scratch.md:
````diff
@@ -1,108 +1,1150 @@
+     Address: {679fed8f4f3ea421689136b25073c6da7973418f}
+    ```
+
+    !!! note 
````
This is a bit confusing, maybe something like? It might work better after you introduce `alloc` , e.g. as a note on 4.
**note** You could generate multiple accounts for a single node, or any number of accounts for additional nodes and pre-allocate them with funds in the genesis.json file (see below).

libby commented on docs/Getting Started/Getting-Started-From-Scratch.md:
```diff
@@ -1,108 +1,1150 @@
+
+    !!! note 
+        You could generate one or more accounts (funded) accounts for additional node you want to start in advance and add them to genesis.json file as well
+    
```
The **note** from above could be moved here, and it might be helpful to show `alloc` with two pre-allocated accounts, e.g. 
"`alloc` holds optional accounts with a pre-funded amounts. In this example we are funding the accounts `679fed8f4f3ea421689136b25073c6da7973418f` (generated from the step above) and `c5c7b431e1629fb992eb18a79559f667228cd055` "

libby commented on docs/Getting Started/Getting-Started-From-Scratch.md:
````diff
@@ -1,108 +1,1150 @@
-6. Execute `bootnode --nodekey=new-node-1/nodekey --writeaddress` and take note of the displayed output. This is the enode id of the new node
+    ```
+    $ vim genesis.json
+    ... paste below with account '679fed8f4f3ea421689136b25073c6da7973418f' created at previous step
````
"... alloc account '679fed8f4f3ea421689136b25073c6da7973418f' created in the previous step and an additional account 'c5c7b431e1629fb992eb18a79559f667228cd055`"
...
```
"alloc": {
    "0x679fed8f4f3ea421689136b25073c6da7973418f": {
      "balance": "1000000000000000000000000000"
    },
    "0xc5c7b431e1629fb992eb18a79559f667228cd055": {
      "balance": "1000000000000000000000000000"
    }
  }
```

libby commented on docs/Getting Started/Getting-Started-From-Scratch.md:
````diff
@@ -1,108 +1,1150 @@
+    ```
+6. Execute below command to display enode id of the new node
+    ```
+    $ bootnode --nodekey=new-node-1/nodekey --writeaddress
````
$ bootnode --nodekey=new-node-1/nodekey --writeaddress > new-node-1/enode
$ cat new-node-1/enode
'70399c3d1654c959a02b73acbdd4770109e39573a27a9b52bd391e5f79b91a42d8f2b9e982959402a97d2cbcb5656d778ba8661ec97909abc72e7bb04392ebd8'

libby commented on docs/Getting Started/Getting-Started-From-Scratch.md:
````diff
@@ -1,108 +1,1150 @@
+5. Generate node key and copy it into datadir
+    ```
+    $ bootnode --genkey=nodekey
+    $ cp nodekey new-node-1/
````
```
mv nodekey new-node-1
```

libby commented on docs/Getting Started/Getting-Started-From-Scratch.md:
```diff
@@ -1,108 +1,1150 @@
+    > exit
+    $
+    $
+    $ ps
```
```
ps | grep geth
```

libby commented on docs/Getting Started/Getting-Started-From-Scratch.md:
```diff
@@ -1,108 +1,1150 @@
+    INFO [06-11|16:14:24.833] Persisted trie from memory database      nodes=6 size=1.01kB time=52.828µs gcnodes=0 gcsize=0.00B gctime=0s livenodes=1 livesize=0.00B
+    INFO [06-11|16:14:24.834] Successfully wrote genesis state         database=lightchaindata                                                  hash=b992be…533db7
+    $
+    $ cd ..
```
maybe change the 3 commands to 1 command:
```
cd ../node0 && geth --datadir data init genesis.json
cd ../node1 && geth --datadir data init genesis.json
cd ../node2 && geth --datadir data init genesis.json
```

libby commented on docs/Getting Started/Getting-Started-From-Scratch.md:
```diff
@@ -1,108 +1,1150 @@
+        This configuration starts Quorum without privacy support as could be evidenced in prefix `PRIVATE_CONFIG=ignore`, please see below sections on [how to enable privacy with privacy transaction managers](#adding-privacy-transaction-manager).
+        Please note that istanbul-tools may be used to generate X number of nodes, more information is available in the [docs](https://github.com/jpmorganchase/istanbul-tools).
 
+    Your node is now operational and you may attach to it with `geth attach data/geth.ipc`. 
```
Add a node dir in front of data ```geth attach node0/data/geth.ipc```

libby commented on docs/Getting Started/Getting-Started-From-Scratch.md:
```diff
@@ -1,108 +1,1150 @@
+    cd ../node4
+    PRIVATE_CONFIG=ignore nohup geth --datadir data --nodiscover --istanbul.blockperiod 5 --syncmode full --mine --minerthreads 1 --verbosity 5 --networkid 10 --rpc --rpcaddr 0.0.0.0 --rpcport 22004 --rpcapi admin,db,eth,debug,miner,net,shh,txpool,personal,web3,quorum,istanbul --emitcheckpoints --port 30304 2>>node.log &
+    $
+    $ chmod +x startall.sh
```
Make sure to stop the previous nodes before starting the new set.
```
# see if the any geth nodes are running.
$ ps | grep geth
# kill geth processes
$ killall -INT geth
```

libby commented on docs/Getting Started/Getting-Started-From-Scratch.md:
```diff
@@ -1,108 +1,1150 @@
 
 ## Adding privacy transaction manager
 ### Tessera
 1. Build Quorum and install [Tessera](https://github.com/jpmorganchase/tessera/releases) as described in the [getting set up](../Setup%20Overview%20%26%20Quickstart) section. Ensure that PATH contains geth and bootnode. Be aware of the location of the `tessera.jar` release file
```
[getting set up] link is  broken.

libby commented on docs/Getting Started/Getting-Started-From-Scratch.md:
```diff
@@ -1,108 +1,1150 @@
+    
+    Please re-enter the password (or lack of) to confirm
+    
+    10:45:02.567 [main] INFO  com.quorum.tessera.nacl.jnacl.Jnac
```
The `privateFor` array in the private-contract.js (below) must contain the tm public key of an existing tessera node.
Replace the `privateFor` key below with the key you generated for node2.
```
$ cat new-node-2t/new-node-2.pub
CziOcGtPppFyngwMz9HXyannocfCOgInG2DiJJz8A1s=
## place you key in the privateFor array.
privateFor: ["CziOcGtPppFyngwMz9HXyannocfCOgInG2DiJJz8A1s="]
```
if you do not have an existing public key in the array, you will see the following error when the script in loaded (next step).
```
loadScript("private-contract.js")
err creating contract Error: Non-200 status code: &{Status:400 Bad Request StatusCode:400 Proto:HTTP/1.1 ProtoMajor:1 ProtoMinor:1 Header:map[Date:[Mon, 17 Jun 2019 15:23:53 GMT] Content-Type:[text/plain] Content-Length:[73] Server:[Jetty(9.4.z-SNAPSHOT)]] Body:0xc01997a580 ContentLength:73 TransferEncoding:[] Close:false Uncompressed:false Trailer:map[] Request:0xc019788200 TLS:<nil>}
```

Commit:
Updates quickstart to highlight quorum-examples

The link to quorum-examples and the explanation that it is the recommended way to get started was kind of lost in the middle of a long intro paragraph. This made my eyes jump to Building Quorum Node from Source, thinking it was the first step in getting started. Adding an explicit header and section for Quorum Examples makes it impossible to miss.

Krish1979 committed:
Merge branch 'master' into fromscratch-docs-update

Krish1979 committed:
Update Getting-Started-From-Scratch.md

Incorporated review comments.

Krish1979 committed:
Merge branch 'master' into fromscratch-docs-update
